packages/server/api/controllers/tools.controller.js

```diff
@@ -17,6 +19,26 @@ const getTools = async () => {
     'picture',
     'created_at',
   );
+  const categoriesList = await knex('categories')
```

Comment:
Look into how to do a many-to-many join in SQL instead (using something like an inner join). 

SQL is made for this kind of work, use that instead of writing it in JavaScript which will be much slower!

Reply:
I tried to use join, but I have problem with "group by", it give me an error because of the mode "Only full group by".
and without the group by, I get  some tools twice because the have 2 categories.
I don’t  know how can I fix it.

Reply:
I'd check in with one of the mentors who is more familiar with your SQL setup but this article might help https://dev.mysql.com/doc/refman/5.7/en/group-by-handling.html

We ran into this problem in the class 20 final project as well, and decided to disable this check in the database.